@here/olp-sdk-core/lib/utils/multipartupload-internal/NodeFileData.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import { BlobData, fs } from "@here/olp-sdk-core";
+
+/**
+ * @internal
+ * Implementation of reading bytes from file by filepath
+ * in Node.js.
+ */
+export class NodeFileData implements BlobData {
+    private readonly fileHandle: fs.promises.FileHandle;
+    private readonly fileStats: fs.Stats;
+
+    constructor(opts: {
+        fileHandle: fs.promises.FileHandle;
+        fileStats: fs.Stats;
+    }) {
+        this.fileHandle = opts.fileHandle;
+        this.fileStats = opts.fileStats;
+    }
+
+    static async build(filePath: string): Promise<BlobData> {

```

Comment:
Great solution. But I would call it `fromPath()` so that we can do later on, if needed, `fromHandle()`.

Reply:
renamed, tnx